concent_api/core/validation.py

```diff
@@ -334,3 +336,23 @@ def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) ->
             )
 
     validate_scene_file(extra_data['scene_file'])
+
+
+def validate_if_given_object_is_expected_enum_instance(
+    custom_object: enum.Enum,
+    base_enum: type,
+    allow_none_as_instance: bool = False,
+) -> None:
+    """
+    Function validates, if given object (custom_object) is given Enum instance (base_enum).
+    If it is not, raises TypeError. If given object is None NoneAsValueNotAllowedError will be raised,
+    unless allow_none_as_instance will be True.
+    """
+    if allow_none_as_instance is True and custom_object is None:
+        return
+
+    if not allow_none_as_instance and custom_object is None:
```

Comment:
I think this part would be more readable as:

``` python
if custom_object is None:
    if not allow_none_as_instance:
        raise NoneAsValueNotAllowedError(f'{custom_object} is None, it should be {base_enum} instance')
    else:
        return
```

This immediately shows that these two conditions validate the case where `custom_object` is `None`. In your version you have to pause for a moment to figure that out.

Reply:
You are right. But changing like below this part of code will not be used any more